integration-tests/local-integration-tests/src/statemine.rs

```diff
@@ -0,0 +1,268 @@
+			"balance increased"
+		);
+		assert!(
+			amount > statemine_runtime::Balances::free_balance(&AccountId::from(BOB)),
+			"fee taken"
+		);
+	});
+}
+
+/// Statemine issues custom token
+#[test]
+fn this_chain_statemine_transfers_back_and_forth_work() {
+	simtest();
+	let this_parachain_account: AccountId =
+		polkadot_parachain::primitives::Sibling::from(THIS_PARA_ID).into_account();
+	let this_para_id: AccountId = ParaId::from(THIS_PARA_ID).into_account();
+	let state_mine_para_id: AccountId = ParaId::from(STATEMINE_PARA_ID).into_account();
+
+	// minimum asset should be: FEE_WEIGHT+FEE_KUSAMA+max(KUSAMA_ED,STATEMINE_ED+FEE_STATEMINE).
+	// but due to current half fee, sender asset should at lease: FEE_WEIGHT + 2 * FEE_KUSAMA
+	let relay_native_asset_amount = 3 * FEE_WEIGHT + 3 * FEE_KUSAMA;
+	let remote_asset_id = 3451561; // magic number to avoid zero defaults and easy to find
+	let foreign_asset_id_on_this = register_statemine_asset(remote_asset_id);
+	let accounted_native_balance = statemine_side(TEN + relay_native_asset_amount, remote_asset_id);
+
+	let (this_reserve, statemine_reserve) = KusamaRelay::execute_with(|| {
+		let _ = kusama_runtime::Balances::make_free_balance_be(&this_para_id, TEN);
+		(
+			kusama_runtime::Balances::balance(&this_para_id),
+			kusama_runtime::Balances::balance(&state_mine_para_id),
+		)
+	});
+
+	this_chain_side(relay_native_asset_amount, foreign_asset_id_on_this);
+
+	// during transfer relay rebalanced amounts
+	KusamaRelay::execute_with(|| {
+		assert!(kusama_runtime::Balances::free_balance(&this_para_id) < this_reserve);
+		assert!(statemine_reserve < kusama_runtime::Balances::free_balance(&state_mine_para_id));
+	});
+
+	log::info!(target : "xcmp::test", "checking that assets for Bob are back");
+	Statemine::execute_with(|| {
+		use statemine_runtime::*;
+		// This send back custom asset to Statemine, ensure recipient got custom asset
+		assert_eq!(UNIT, Assets::balance(remote_asset_id, &AccountId::from(BOB)));
+		// and withdraw sibling parachain sovereign account
+		assert_eq!(9 * UNIT, Assets::balance(remote_asset_id, &this_parachain_account));
+
+		assert_eq!(
+			1003989333336, // approx. UNIT + FEE_WEIGHT - FEE_STATEMINE,
+			Balances::free_balance(&AccountId::from(BOB))
+		);
+		let new_balance = Balances::free_balance(&this_parachain_account);
+		assert!(accounted_native_balance <= new_balance);
+		assert_eq!(
+			10016522666636, /* approximately this UNIT + asset_amount - FEE_WEIGHT - FEE_KUSAMA
+			                 * - FEE_STATEMINE - FEE_WEIGHT, */
+			new_balance,
+		);
+	});
+}
+
+// transfer custom asset from this chain  to Statemine
+fn this_chain_side(fee_amount: u128, foreign_asset_id_on_this: CurrencyId) {
+	This::execute_with(|| {
+		use this_runtime::*;
+
+		let bob_statemine_asset_amount =
+			Tokens::free_balance(foreign_asset_id_on_this, &AccountId::from(BOB));
+		assert_eq!(9_999_936_000_000, bob_statemine_asset_amount,);
```

Comment:
If I understand correctly this is `TEN` - fee ?

Reply:
Maybe yes. so it is strait port from Acala (they sometimes has calculation and sometimes hardcode), will improve later (need to get exact fee values from other runtimes)